From e99a3469ccd31e74eab20a867c3922d88e197720 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Olivier=20De=20Canni=C3=A8re?= <olivier.decanniere@qt.io>
Date: Tue, 21 Oct 2025 13:55:41 +0200
Subject: [PATCH 189/239] qmllint: Fix typo in assignmentInCondition warning

Amends ce69cc78c0aaa0afea5cd74d15ea37685948e6d0

Change-Id: I43b514bc390f84151814eb74b3062c80162218c3
Reviewed-by: Fabian Kosmale <fabian.kosmale@qt.io>
(cherry picked from commit 1727532ff9f86ccd49d428cfdc1a64044bb6f044)
Reviewed-by: Qt Cherry-pick Bot <cherrypick_bot@qt-project.org>
---
 src/qmlcompiler/qqmljsimportvisitor.cpp | 2 +-
 tests/auto/qml/qmllint/tst_qmllint.cpp  | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/qmlcompiler/qqmljsimportvisitor.cpp b/src/qmlcompiler/qqmljsimportvisitor.cpp
index b6006d05fd..ec6c573ec5 100644
--- a/src/qmlcompiler/qqmljsimportvisitor.cpp
+++ b/src/qmlcompiler/qqmljsimportvisitor.cpp
@@ -3300,7 +3300,7 @@ bool QQmlJSImportVisitor::visit(IfStatement *statement)
     if (BinaryExpression *binary = cast<BinaryExpression *>(statement->expression)) {
         if (binary->op == QSOperator::Assign) {
             m_logger->log(
-                    "Assignment in condition: did you meant to use \"===\" or \"==\" instead of \"=\"?"_L1,
+                    "Assignment in condition: did you mean to use \"===\" or \"==\" instead of \"=\"?"_L1,
                     qmlAssignmentInCondition, binary->operatorToken);
         }
     }
diff --git a/tests/auto/qml/qmllint/tst_qmllint.cpp b/tests/auto/qml/qmllint/tst_qmllint.cpp
index 763663cd85..3cf73d6d86 100644
--- a/tests/auto/qml/qmllint/tst_qmllint.cpp
+++ b/tests/auto/qml/qmllint/tst_qmllint.cpp
@@ -1606,7 +1606,7 @@ void TestQmllint::dirtyJsSnippet_data()
 
     QTest::newRow("assignmentInCondition")
             << u"let xxx = 3; if (xxx=3) return;"_s
-            << Result{ { { "Assignment in condition: did you meant to use \"===\" or \"==\" "
+            << Result{ { { "Assignment in condition: did you mean to use \"===\" or \"==\" "
                            "instead of \"=\"?"_L1,
                            1, 21 } } }
             << defaultOptions;
-- 
2.51.2

